### PR TITLE
decomp: Pass of the final remaining gameplay related files

### PR DESCRIPTION
**Pull request by xTVaser**
Doing a pass of the final gameplay/level code that hasn't been touched yet to find remaining issues / hopefully finish them off.

Files remaining:
- [x] target-death
  - do in https://github.com/water111/jak-project/pull/893
- [x] billy
- [x] collide-reaction-racer
  - do in https://github.com/water111/jak-project/pull/893
- [x] pelican
- [x] target2
  - do in https://github.com/water111/jak-project/pull/893
- [x] citadel-sages
- [x] citb-bunny
- [ ] snow-bunny
  - blocked
- [ ] lavatube-energy
  - blocked
- [ ] collectables
  - blocked
- [x] target-handler
  - do in https://github.com/water111/jak-project/pull/893
- [x] lurkerworm
- [ ] memory-usage
  - blocked
- [ ] joint-exploder
  - partial
- [x] default-menu
- [x] floating-launcher
- [x] shover
- [x] steam-cap
- [x] sun-iris-door
- [x] whirlpool

### CODE REVIEW

File: goal_src/levels/finalboss/green-eco-lurker.gc [hunk 1]
```diff
@@ -40,7 +40,7 @@
   0
   -1
   ((1 (meters 999999)))
-  :bounds (static-spherem 0 0 0 5.95)
+  :bounds (static-spherem 0 0 0 5.949999809265137)
```
Comment (water111) [inline goal_src/levels/finalboss/green-eco-lurker.gc:43]: hm, something has gone wrong here... I bet we forgot to use the float printer for the skelgroup definition.  fine for now though

File: goal_src/levels/finalboss/green-eco-lurker.gc [hunk 2]
```diff
@@ -467,9 +470,15 @@
...
-      (eval-path-curve-div! (-> obj path) arg0 (the float s2-0) 'interp)
-      (if (dummy-51 obj arg0)
-       (return #t)
+      (let* ((a0-2 (-> obj path))
+             (t9-1 (method-of-object a0-2 eval-path-curve-div!))
+             (a1-1 arg0)
+             (a2-0 (the float s2-0))
+             )
+       (t9-1 a0-2 a1-1 a2-0 'interp)
+       (if (dummy-51 obj arg0 (the-as vector a2-0))
```
Comment (water111) [inline goal_src/levels/finalboss/green-eco-lurker.gc:479]: I don't think this method actually use `a2-0` as an argument. Maybe we need to use `replace` on this method too

File: goal_src/levels/maincave/baby-spider.gc [hunk 3]
```diff
@@ -1189,18 +1196,23 @@ baby-spider-default-event-handler
...
+    (the-as
+     (function object object object object object)
+     cavecrystal-light-control-default-callback
+     )
+    -1
+    4096.0
     )
-   -1
-   4096.0
+   (TODO-RENAME-48 obj (the-as object a1-4))
```
Comment (water111) [inline goal_src/levels/maincave/baby-spider.gc:1214]: I don't think this takes `a1-4` as an argument

File: goal_src/levels/sunken/shover.gc [hunk 4]
```diff
@@ -5,3 +5,198 @@
...
+  (behavior ((arg0 process) (arg1 int) (arg2 symbol) (arg3 event-message-block))
+   (case arg2 
+    (('touch 'attack)
+      (when (= (-> arg0 type) target)
+       (-> arg3 param 0)
+       (let ((t9-0 (method-of-type touching-shapes-entry dummy-13)))
+        (-> *target* control)
+        1
+        0
+        (when (t9-0)
```
Comment (water111) [inline goal_src/levels/sunken/shover.gc:46]: I think this takes 3 arguments

File: goal_src/levels/sunken/shover.gc [hunk 5]
```diff
@@ -5,3 +5,198 @@
...
+         (let ((s4-0 (new 'stack 'attack-info)))
+          (let ((t9-2 (method-of-object (-> self root-override) dummy-41)))
+           (-> self shove-up)
+           (t9-2)
```
Comment (water111) [inline goal_src/levels/sunken/shover.gc:50]: this should have arguments

File: goal_src/levels/sunken/shover.gc [hunk 6]
```diff
@@ -5,3 +5,198 @@
...
+       )
+      )
+    )
+   )
+  :code
+  (the-as (function none :behavior shover) anim-loop)
+  )
+
+(defmethod copy-defaults! shover ((obj shover) (arg0 res-lump))
+  (local-vars (sv-16 int))
```
Comment (water111) [inline goal_src/levels/sunken/shover.gc:115]: needs to have type `res-tag`

File: goal_src/levels/sunken/steam-cap.gc [hunk 7]
```diff
@@ -5,3 +5,996 @@
...
+   )
+  (if (nonzero? (-> obj part3))
+   (kill-and-free-particles (-> obj part3))
+   )
+  ((method-of-type process-drawable deactivate) obj)
+  (none)
+  )
+
+(defmethod copy-defaults! steam-cap ((obj steam-cap) (arg0 res-lump))
+  (local-vars (sv-16 int))
```
Comment (water111) [inline goal_src/levels/sunken/steam-cap.gc:878]: needs to have type `res-tag`

File: goal_src/levels/sunken/sun-iris-door.gc [hunk 8]
```diff
@@ -5,3 +5,600 @@
...
+     (cond
+      ((and *target* (-> obj directional-proximity?))
+       (let ((s4-6 (new 'stack-no-clear 'vector))
+             (s5-3 (new 'stack-no-clear 'vector))
+             )
+        (vector-! s4-6 (target-pos 0) (-> obj root-override trans))
+        (set! (-> s4-6 y) 0.0)
+        (vector-! s5-3 (camera-pos) (-> obj root-override trans))
+        (set! (-> s5-3 y) 0.0)
+        (case (>= (vector-dot (-> obj outward-vec) s4-6) 0.0)
```
Comment (water111) [inline goal_src/levels/sunken/sun-iris-door.gc:268]: this is weird, I think it's the first case where the decompiler rewrote something as a `case` that probably wasn't originally a case.

File: goal_src/levels/sunken/sun-iris-door.gc [hunk 9]
```diff
@@ -5,3 +5,600 @@
...
+    (else
+     (transform-post)
+     )
+    )
+   (none)
+   )
+  )
+
+(defmethod copy-defaults! sun-iris-door ((obj sun-iris-door) (arg0 res-lump))
+  (local-vars (sv-16 int))
```
Comment (water111) [inline goal_src/levels/sunken/sun-iris-door.gc:437]: needs to be a `res-tag`

File: goal_src/levels/sunken/whirlpool.gc [hunk 10]
```diff
@@ -5,3 +5,513 @@
...
+       (-> s4-1 z)
+       (* (- (-> s4-1 z) (-> gp-0 z)) (-> *display* frames-per-second))
+       )
+      )
+     (let ((gp-1 (-> *target* control)))
+      (send-event (-> gp-1 process) 'rotate-y-angle f30-0)
+      (let ((t9-6 (method-of-object gp-1 dummy-32)))
+       8192.0
+       (-> gp-1 root-prim collide-with)
+       (t9-6)
```
Comment (water111) [inline goal_src/levels/sunken/whirlpool.gc:367]: this is missing args

File: goal_src/levels/sunken/whirlpool.gc [hunk 11]
```diff
@@ -5,3 +5,513 @@
...
+       )
+      (let ((s3-0 (-> gp-1 status)))
+       ((method-of-object gp-1 dummy-37))
```
Comment (water111) [inline goal_src/levels/sunken/whirlpool.gc:370]: missing args

File: goal_src/levels/sunken/whirlpool.gc [hunk 12]
```diff
@@ -5,3 +5,513 @@
...
+(defmethod deactivate whirlpool ((obj whirlpool))
+  (if (nonzero? (-> obj sound))
+   (stop! (-> obj sound))
+   )
+  ((method-of-type process-drawable deactivate) obj)
+  (none)
+  )
+
+(defmethod copy-defaults! whirlpool ((obj whirlpool) (arg0 res-lump))
+  (local-vars (sv-16 int))
```
Comment (water111) [inline goal_src/levels/sunken/whirlpool.gc:438]: needs to be `res-tag`

File: goal_src/levels/swamp/billy.gc [hunk 13]
```diff
@@ -5,3 +5,2048 @@
...
+  ((arg0 billy) (arg1 vector) (arg2 vector))
+  (set! (-> self billy) (the-as (pointer billy) (process->ppointer arg0)))
+  (set! (-> self destination quad) (-> arg2 quad))
+  (set! (-> self dest-type) (the-as uint 1))
+  (let ((s4-0 (new 'stack-no-clear 'vector)))
+   (vector-! s4-0 arg2 arg1)
+   (set! (-> s4-0 y) 0.0)
+   (vector-normalize! s4-0 1.0)
+   (swamp-rat-init-by-other
+    (the-as swamp-rat-nest arg0)
```
Comment (water111) [inline goal_src/levels/swamp/billy.gc:527]: I don't think this cast is right

File: goal_src/levels/swamp/billy.gc [hunk 14]
```diff
@@ -5,3 +5,2048 @@
...
+                            )
+                           )
+                     (when v1-2
+                      (set!
+                       (-> (the-as billy-rat v1-2) dest-type)
+                       (the-as uint 1)
+                       )
+                      (get-random-point
+                       (-> self path-waypts)
+                       (the-as vector (&-> v1-2 stack 400))
```
Comment (water111) [inline goal_src/levels/swamp/billy.gc:1012]: hm, this is weird, are we sure about this type?

File: goal_src/levels/swamp/swamp-rat.gc [hunk 15]
```diff
@@ -870,11 +870,16 @@ swamp-rat-default-event-handler
...
-  (TODO-RENAME-48 self)
+  (let ((t9-1 forward-up->quaternion)
+        (a0-4 (-> self collide-info quat))
+        (a1-1 arg2)
+        )
+   (t9-1 a0-4 a1-1 *up-vector*)
+   (vector-float*! (-> self collide-info scale) *identity-vector* 1.5)
+   (vector-float*! (-> self collide-info transv) arg2 49152.0)
+   (set! (-> self entity) (-> arg0 entity))
+   (TODO-RENAME-48 self (the-as object a1-1))
```
Comment (water111) [inline goal_src/levels/swamp/swamp-rat.gc:881]: don't think this takes `a1-1` as an argument

File: goal_src/levels/finalboss/green-eco-lurker.gc (hunk 1 above)
Comment (ManDude) [reply goal_src/levels/finalboss/green-eco-lurker.gc:43]: oops, it doesn't.

File: goal_src/levels/swamp/billy.gc (hunk 14 above)
Comment (xTVaser) [reply goal_src/levels/swamp/billy.gc:1012]: missed a cast

File: goal_src/levels/sunken/steam-cap.gc (hunk 7 above)
Comment (water111) [reply goal_src/levels/sunken/steam-cap.gc:878]: still needs to be `res-tag`

**Pull request merged**